shared/lib/multichain.ts

```diff
@@ -1,5 +1,15 @@
 import { validate, Network } from 'bitcoin-address-validation';
 
+/**
+ * Multi-chain family type.
+ */
+export enum MultichainType {
```

Comment:
I'm not sure if `MultichainType` is the right name here, something like `Namespace` or `Caip2Namespace` would make more sense. However, we should pay attention to not reuse a type already used in `@metamask/utils`.

Reply:
I initially wanted to use the one coming from `@metamask/utils` but we have no `Bip122` yet 😅

I can try to update this too, I started a PR some days ago here:
- https://github.com/MetaMask/utils/pull/213

Reply:
I'm now using `KnownCaipNamespace` for this type instead!